src/main/java/org/opensearch/geospatial/ip2geo/action/DeleteDatasourceTransportAction.java

```diff
@@ -111,9 +111,18 @@ protected void deleteDatasource(final String datasourceName) throws IOException
         if (datasource == null) {
             throw new ResourceNotFoundException("no such datasource exist");
         }
-
+        DatasourceState previousState = datasource.getState();
         setDatasourceStateAsDeleting(datasource);
-        geoIpDataDao.deleteIp2GeoDataIndex(datasource.getIndices());
+
+        try {
+            geoIpDataDao.deleteIp2GeoDataIndex(datasource.getIndices());
+        } catch (Exception e) {
+            if (previousState.equals(datasource.getState()) == false) {
```

Comment:
what is the scenario when datasource has previous state? We're setting it to "deleting" before we call delete method in setDatasourceStateAsDeleting().

Reply:
We first change the datasource state to "deleting" and run the delete process. So, datasource always has previous state either "available", "create_failed", or "deleting".

Reply:
Got more info on scenario, we need to delete two resources(index and metadata), if we fail to delete metadata, we don’t revert state because index for data is already deleted